dpnp/backend_mathematical.pyx

```diff
@@ -154,6 +155,26 @@ cpdef dparray dpnp_cumsum(dparray x1):
     return result
 
 
+cpdef dparray dpnp_diff(dparray input):
+    size_i = input.size
+    shape_i = input.shape
```

Comment:
The same as comment with `size_i` variable.
The new variable with unclear name and purpose is defined. It is hard enough to read what happens with it below in code, at least for me. I suggest to avoid such things.